lib/data_stores/mysql_advanced_data_store.py

```diff
@@ -26,23 +26,48 @@ class Error(data_store.Error):
   """Base class for all exceptions in this module."""
 # pylint: enable=nonstandard-exception
 
+class SafeQueue(Queue.Queue):
+  def __init__(self, maxsize=0):
+      self.maxsize = maxsize
+      self._init(maxsize)
```

Comment:
If you look at the other methods of Queue and the subclasses it makes more sense as to why they did it that way